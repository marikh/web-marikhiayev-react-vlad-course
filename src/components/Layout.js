import React from 'react';
import { 
    Header,
    Body,
    Footer,
    SocialLinks
} from '../components'

import { Route } from 'react-router-dom';
import classNames from 'classnames';
import T from 'i18n-react';

const socialLinks = [
  {
    text: 'Facebook',
    icon: 'facebook',
    link: 'facebook.com'
  },
  {
    text: 'Github',
    icon: 'github',
    link: 'github.com'
  },
  {
    text: 'Email',
    icon: 'envelope',
    link: 'vnovick@gmail.comm'
  }
]

export default class Layout extends React.Component {
  state = {
    menuState: false
  }

  open = () =>  {
    this.setState({
      menuState: true
    })
  }

  
  close = (e) => {
    if (this.state.menuState) {
      this.setState({
        menuState: false
      })
    } 
  }

  render() {
    const { 
        open, 
        close, 
        state: { menuState }, 
        props: { 
            heroContent: HeroArea,
            className,
            footerContent: FooterContent,
            children
        }
    } = this;
    return (
        <div className={classNames(["App", className ])} onClick={() => close()} >    
             <Header menuConfig={{ menuState, open, socialLinks }}>
                  { HeroArea && <HeroArea links={socialLinks}/> }
              </Header>
              <Body className={classNames([T.translate("LangDirection") === "rtl" && "rtl"])}>
                  { children }
              </Body>
              <Footer className={classNames([T.translate("LangDirection") === "rtl" && "rtl"])}>
                  { FooterContent ? <FooterContent links={socialLinks}/> : <SocialLinks links={socialLinks} />} 
              </Footer>
        </div>
    )
  }
}