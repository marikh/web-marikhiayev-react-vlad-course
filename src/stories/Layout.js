import React from 'react';
import { 
    Header,
    Body,
    Footer,
    SocialLinks
} from '../components'

import { Route } from 'react-router-dom';
import classNames from 'classnames';

const socialLinks = [
  {
    text: 'Facebook',
    icon: 'facebook',
    link: 'facebook.com'
  },
  {
    text: 'Github',
    icon: 'github',
    link: 'github.com'
  },
  {
    text: 'Email',
    icon: 'envelope',
    link: 'vnovick@gmail.comm'
  }
]

export default class Layout extends React.Component {
  state = {
    menuState: false
  }

  open = () =>  {
    this.setState({
      menuState: true
    })
  }

  
  close = (e) => {
    if (this.state.menuState) {
      this.setState({
        menuState: false
      })
    } 
  }

  render() {
    const { 
        open, 
        close, 
        state: { menuState }, 
        props: { 
            heroContent: HeroArea,
            className,
            footerContent: FooterContent,
            children
        }
    } = this;
    return (
        <div className={classNames(["App", className ])} onClick={() => close()} >    
             <Header menuConfig={{ menuState, open, socialLinks }}>
                  { HeroArea && <HeroArea links={socialLinks}/> }
              </Header>
              <Body>
                  { children }
              </Body>
              <Footer>
                  { FooterContent ? <FooterContent links={socialLinks}/> : <SocialLinks links={socialLinks} />} 
              </Footer>
        </div>
    )
  }
}