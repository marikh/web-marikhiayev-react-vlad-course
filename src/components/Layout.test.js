import Layout from './Layout';
import { mount, shallow, dive } from "enzyme";
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import { Header, Body, Footer, Menu } from './';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';

const INITIAL_STATE = {
        showProtectedLinks : false
    };

const mockedStore = createMockStore({ menu : INITIAL_STATE });

describe('Layout outer tests', () => {
    let props;
    let mountedlayoutComponent;

    const layoutComponent = () => {
        if (!mountedlayoutComponent) {
            mountedlayoutComponent = mount(
                <Provider store={mockedStore}>
                    <MemoryRouter>
                        <Layout  />
                    </MemoryRouter>
                </Provider>
            );
        }
        return mountedlayoutComponent
    }

    beforeEach(() => {
        props = {};
        if(mountedlayoutComponent != null){
            mountedlayoutComponent.unmount();
            mountedlayoutComponent = undefined;
        }
        
    });

    it("always renders a section", () => {

        const sections = layoutComponent().find("section");
        expect(sections.length).toBeGreaterThan(0);
    });

    it("has Header", () => {
        expect(layoutComponent().find(Header).length).toBeGreaterThan(0)
    })
    
    it("has Body Component", () => {
        expect(layoutComponent().find(Body).length).toBeGreaterThan(0)
    })

    it("has Footer component", () => {
        expect(layoutComponent().find(Footer).length).toBeGreaterThan(0)
    })

    it('Layout triggers menu open when clicked on menu', () => {
        layoutComponent().find(Menu).find('.hamburger-icon').simulate('click')
        expect(layoutComponent().find(Menu).props().menuConfig.menuState).toBeTruthy();
    })
})

describe('Layout snapshot tests', () => {
    it('test weird snapshot', () => {

        const tree = renderer.create(
            <Provider store={mockedStore}>
                <MemoryRouter>
                    <Layout  />
                </MemoryRouter>
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
 })

describe('Layout shallow tests', () => {

    it(`Layout has menu state and it's false`, () => {
        const layout = shallow(<Layout />)
        expect(layout.state().menuState).toBe(false);
    })

    it(`Layout manages menu close on click`, () => {
        const layout = shallow(<Layout />)
        layout.state().menuState = true;
        expect(
            layout.simulate('click')
                .state()
                    .menuState
        ).toBe(false);
    })

})


