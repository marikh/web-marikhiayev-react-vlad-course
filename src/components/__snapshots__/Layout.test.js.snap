// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot tests test weird snapshot 1`] = `
<div
  className="App"
  onClick={[Function]}
>
  <header
    className="App-header"
  >
    <div
      className="App-menubar"
    >
      <label
        className="hamburger-icon fa fa-bars"
        onClick={[Function]}
      />
      <div
        className="menu-content"
      >
        <div
          className="links-container"
        >
          <a
            className="page-link"
            href="/"
            onClick={[Function]}
          >
            <span>
              Home
            </span>
          </a>
          <a
            className="page-link"
            href="/about"
            onClick={[Function]}
          >
            <span>
              About
            </span>
          </a>
          <a
            className="page-link"
            href="/products"
            onClick={[Function]}
          >
            <span>
              Products
            </span>
          </a>
          <a
            className="page-link"
            href="/contact"
            onClick={[Function]}
          >
            <span>
              Contact
            </span>
          </a>
          <a
            className="page-link"
            href="/login"
            onClick={[Function]}
          >
            <span>
              SignIn
            </span>
          </a>
        </div>
        <div
          className="socialLinks"
        >
          <div
            className="social-links"
          >
            <a
              className="fa fa-facebook"
              href="facebook.com"
              target="_blank"
            />
            <a
              className="fa fa-github"
              href="github.com"
              target="_blank"
            />
            <a
              className="fa fa-envelope"
              href="vnovick@gmail.comm"
              target="_blank"
            />
          </div>
        </div>
      </div>
      <div
        className="overlay"
      />
    </div>
    <section
      className="App-subheader"
    />
  </header>
  <section
    className="App-body"
  />
  <footer
    className="App-footer"
  >
    <div
      className="social-links"
    >
      <a
        className="text-link"
        href="facebook.com"
        target="_blank"
      >
        Facebook
      </a>
      <a
        className="text-link"
        href="github.com"
        target="_blank"
      >
        Github
      </a>
      <a
        className="text-link"
        href="vnovick@gmail.comm"
        target="_blank"
      >
        Email
      </a>
    </div>
  </footer>
</div>
`;
